pkg/agent/proxy/proxier.go

```diff
@@ -452,19 +452,29 @@ func (p *proxier) installServices() {
 				continue
 			}
 
-			// Install another group when Service externalTrafficPolicy is Local.
-			if p.proxyAll && svcInfo.NodeLocalExternal() {
+			if p.proxyAll {
 				groupIDLocal, _ := p.groupCounter.Get(svcPortName, true)
-				var localEndpointList []k8sproxy.Endpoint
-				for _, ed := range endpointUpdateList {
-					if !ed.GetIsLocal() {
+				// Install another group when Service externalTrafficPolicy is Local.
+				if svcInfo.NodeLocalExternal() {
+					var localEndpointList []k8sproxy.Endpoint
+					for _, ed := range endpointUpdateList {
+						if !ed.GetIsLocal() {
+							continue
+						}
+						localEndpointList = append(localEndpointList, ed)
+					}
+					if err = p.ofClient.InstallServiceGroup(groupIDLocal, svcInfo.StickyMaxAgeSeconds() != 0, localEndpointList); err != nil {
+						klog.ErrorS(err, "Error when installing Group for Service whose externalTrafficPolicy is Local")
 						continue
 					}
-					localEndpointList = append(localEndpointList, ed)
-				}
-				if err = p.ofClient.InstallServiceGroup(groupIDLocal, svcInfo.StickyMaxAgeSeconds() != 0, localEndpointList); err != nil {
-					klog.ErrorS(err, "Error when installing Group for Service whose externalTrafficPolicy is Local")
-					continue
+					// Uninstall the group with only local Endpoints when Service externalTrafficPolicy is changed from Local
+					// to Cluster.
+				} else if !svcInfo.NodeLocalInternal() && pSvcInfo != nil && pSvcInfo.NodeLocalExternal() {
```

Comment:
```suggestion
				} else if !svcInfo.NodeLocalExternal() && pSvcInfo != nil && pSvcInfo.NodeLocalExternal() {
```
But doesn't "else if" already mean it?